Update checks to remove redundancy

JoshStern opened the pull request:
Some of these checks do not always need to be recursive. By cleaning
them and removing the unnecessary hack blocks we no longer generate
unused imports

- Prettier fixes
- JSDoc comments
- Updates identifiersForStatements by adding an explicit recursive flag
and removes unnecessary hack. Restructure loop slightly
- Updates includes generator to allow for explicit Int64 vs thrift
checks to avoid unnecessary imports. Also adds explicit recursion flag
because we only need the recursive walk in a single case
- Adds recursive case to user.thrift integration tests

Commit:
Update checks to remove redundancy

Some of these checks do not always need to be recursive. By cleaning
them and removing the unnecessary hack blocks we no longer generate
unused imports

- Prettier fixes
- JSDoc comments
- Updates identifiersForStatements by adding an explicit recursive flag
and removes unnecessary hack. Restructure loop slightly
- Updates includes generator to allow for explicit Int64 vs thrift
checks to avoid unnecessary imports. Also adds explicit recursion flag
because we only need the recursive walk in a single case
- Adds recursive case to user.thrift integration tests

josephschmitt commented on src/main/render/shared/includes.ts:
```diff
@@ -13,24 +13,38 @@ import { Resolver } from '../../resolver'
 function fieldTypeUsesThrift(
     fieldType: FieldType,
     state: IRenderState,
+    recursiveResolve: boolean = false,
```
I don't know how much we want to stray from the upstream project, but once we get to 3 or 4 params, especially with some that have optional values, it's usually better to use an options with named params instead of a parameter list. Maybe we can do that for at least these two params, and leave fieldType and state as is?

```
function fieldTypeUsesThrift(fieldType: FieldType, state: IRenderState, {
  recursiveResolve, 
  int64Check
} = {
  recursiveResolve: true, 
  int64Check: false
}) {
}
```

That way you only provide the values that stray from the default, like on line #157:
```
return fieldTypeUsesThrift(statement.fieldType, state, {int64Check: true})
```
Instead of 
```
return fieldTypeUsesThrift(statement.fieldType, state, true, true)
```

JoshStern replied:
That sounds good to me. At the very least it keeps our new parameters separate. There are a few parts of this that are safe to contribute back to the original project.

There will definitely be drift but I actually think it will be minor after these changes. I don't plan on touching this again until we run into an error. I've thought of a few cases where struct constants have embedded unions may fail but it'll be easy to expand in that case (adding an `||` to some if checks).

Commit:
Remove vscode directory from gitignore

I initially added it there because I thought it wasn't part of the
original repo. It's already tracking the settings and it would
be useful to share my launch configuration.
